tests/mocks/MockThreadPool.hh

```diff
@@ -52,6 +52,20 @@ public:
             fTask();
     }
 
+    /// <summary>
+    /// Gets the delay until the next task would be executed.
+    /// </summary>
+    std::chrono::milliseconds NextTaskDelay() const
+    {
+        if (!m_vTasks.empty())
+            return std::chrono::milliseconds(0);
+
+        if (!m_vDelayedTasks.empty())
+            return m_vDelayedTasks.front().nDelay;
+
+        return std::chrono::hours(10000);
```

Comment:
Do duration types do implicit casting?

Reply:
Yes. If this code is reached, it will generate an error like this:
```
Message: Assert failed. Expected:<500ms> Actual:<36000000000ms>
```